pennylane_qiskit/qiskit_device2.py

```diff
@@ -275,9 +254,7 @@ def __init__(
         self._init_kwargs = kwargs
         # _kwargs are used instead of the Options for performing raw sample based measurements (using old Qiskit API)
         # the _kwargs are a combination of information from Options and _init_kwargs
-        self._kwargs = None
-        if self.options.simulator.noise_model:
-            self.backend.set_options(noise_model=self.options.simulator.noise_model)
+        self._kwargs = kwargs
```

Comment:
What is the difference between `_init_kwargs` and `_kwargs`? Do we need some validation for it, or can we just rely on the pipeline to raise sufficiently descriptive errors for unrecognized arguments?

Reply:
I'm deleting `_init_kwargs` in #547 because I don't think it's relevant anymore. In the past we needed `_init_kwargs` because we had the option to change kwargs post-device-initialization. However, after discussion with tom, this isn't really what PL devices do so we are opting to not allow users to change options after device initialization. If they want to, for example, change the optimization_level to 1 instead of 2, then they will have to redefine their device rather than write something like `device.options.optimization_level = 2`.